controllers/handler/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -289,10 +291,48 @@ func (r *NodeNetworkConfigurationPolicyReconciler) initializeEnactment(policy nm
 		enactmentConditions.Reset()
 	}
 
-	return &enactment.Status.Conditions, enactmentstatus.Update(r.APIClient, enactmentKey, func(status *nmstateapi.NodeNetworkConfigurationEnactmentStatus) {
+	return &enactment, nil
+}
+
+func (r *NodeNetworkConfigurationPolicyReconciler) fillInEnactmentStatus(policy nmstatev1.NodeNetworkConfigurationPolicy,
```

Comment:
Thinking about this, the Update function is already retrying so it does not make sense to Reconcile if Update fail, so if we just return the error if update or the other stuff fails and we read the NNS outside we can do the following changing the function:

```golang
func (r *NodeNetworkConfigurationPolicyReconciler) fillInEnactmentStatus(nns, nncp, nnce, enactmentConditions) error {...}      
```

And at the caller

```golang

nns, err := r.readNNS(nodeName)
if err != nil {
    ctr.Result{}, err
}

if err := r.fillInEnactmentStatus(nns, *instance, *enactmentInstance, enactmentConditions); err != nil {
    r.Log.Error(err, "failed filling in the NNCE status")
    return ctrl.Result{}, nil
}

```

Reply:
done.